src/SeisSol.h

```diff
@@ -147,200 +144,138 @@ class SeisSol
-	{
-		return m_freeSurfaceIntegrator;
-	}
-
-	writer::FreeSurfaceWriter& freeSurfaceWriter()
-	{
-		return m_freeSurfaceWriter;
-	}
-
-	writer::AnalysisWriter& analysisWriter()
-	{
-		return m_analysisWriter;
-	}
-
-	/** Get the post processor module
-         */
-         writer::PostProcessor& postProcessor()
-         {
-                 return m_postProcessor;
-         }
-
-	io::AsyncIO& asyncIO()
-	{
-		return m_asyncIO;
-	}
-
-	/**
-	 * Get the wave field writer module
-	 */
-	writer::WaveFieldWriter& waveFieldWriter()
-	{
-		return m_waveFieldWriter;
-	}
-
-	/**
-	 * Get the fault writer module
-	 */
-	writer::FaultWriter& faultWriter()
-	{
-		return m_faultWriter;
-	}
-
-	/**
-	 * Get the receiver writer module
-	 */
-	writer::ReceiverWriter& receiverWriter()
-	{
-		return m_receiverWriter;
-	}
+  public:
+  const parallel::Pinning& getPinning() { return pinning; }
+
+  /**
+   * Cleanup data structures
+   */
+  virtual ~SeisSol() { delete m_meshReader; }
+
+  /**
+   * Initialize C++ part of the program
+   */
+  bool init(int argc, char* argv[]);
+
+  /**
+   * Finalize SeisSol
+   */
+  void finalize();
+
+  const char* parameterFile() const { return m_parameterFile.c_str(); }
+
+  initializers::time_stepping::LtsLayout& getLtsLayout() { return m_ltsLayout; }
+
+  initializers::MemoryManager& getMemoryManager() { return *(m_memoryManager.get()); }
+
+  time_stepping::TimeManager& timeManager() { return m_timeManager; }
+
+  Simulator& simulator() { return m_simulator; }
+
+  checkpoint::Manager& checkPointManager() { return m_checkPointManager; }
+
+  sourceterm::Manager& sourceTermManager() { return m_sourceTermManager; }
+
+  solver::FreeSurfaceIntegrator& freeSurfaceIntegrator() { return m_freeSurfaceIntegrator; }
+
+  writer::FreeSurfaceWriter& freeSurfaceWriter() { return m_freeSurfaceWriter; }
+
+  writer::AnalysisWriter& analysisWriter() { return m_analysisWriter; }
+
+  /** Get the post processor module
+   */
+  writer::PostProcessor& postProcessor() { return m_postProcessor; }
+
+  io::AsyncIO& asyncIO() { return m_asyncIO; }
+
+  /**
+   * Get the wave field writer module
+   */
+  writer::WaveFieldWriter& waveFieldWriter() { return m_waveFieldWriter; }
+
+  /**
+   * Get the fault writer module
+   */
+  writer::FaultWriter& faultWriter() { return m_faultWriter; }
+
+  /**
+   * Get the receiver writer module
+   */
+  writer::ReceiverWriter& receiverWriter() { return m_receiverWriter; }
 
   /**
    * Get the energy writer module
    */
-   writer::EnergyOutput& energyOutput() {
-     return m_energyOutput;
-   }
+  writer::EnergyOutput& energyOutput() { return m_energyOutput; }
 
   /**
    * Get the flop counter
    */
-  monitoring::FlopCounter& flopCounter() {
-    return m_flopCounter;
+  monitoring::FlopCounter& flopCounter() { return m_flopCounter; }
+
+  /**
+   * Set the mesh reader
+   */
+  void setMeshReader(seissol::geometry::MeshReader* meshReader) {
+    if (m_meshReader != 0L)
```

Comment:
`!= nullptr`

(maybe longer-term—not necessarily this PR—make a managed pointer?)